cmd/performance-profile-creator/cmd/root.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	powerConsumptionMode, outputFilePath, mustGatherDirPath, profileName string
+	reservedCPUCount                                                     int
+	splitCPUsAcrossNUMA, disableHT, rtKernel, userLevelNetworking        bool
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "performance-profile-creator",
+	Short: "A tool that automates creation of Performance Profiles",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().IntVarP(&reservedCPUCount, "reserved-cpu-count", "R", 0, "Number of reserved CPUs")
+	rootCmd.PersistentFlags().BoolVarP(&splitCPUsAcrossNUMA, "split-cpus-across-numa", "S", true, "Split the CPUs across NUMA nodes")
+	rootCmd.PersistentFlags().BoolVarP(&disableHT, "disable-ht", "H", false, "Disable Hyperthreading")
+	rootCmd.PersistentFlags().BoolVarP(&rtKernel, "rt-kernel", "K", true, "Enable Real Time Kernel")
+	rootCmd.PersistentFlags().BoolVarP(&userLevelNetworking, "user-level-networking", "U", false, "Run with User level Networking(DPDK) enabled")
+	rootCmd.PersistentFlags().StringVarP(&powerConsumptionMode, "power-consumption-mode", "P", "cstate", "The power consumption mode")
+	rootCmd.PersistentFlags().StringVarP(&outputFilePath, "output-file-path", "O", "/output/sample-profile.json", "Performance Profile file output path")
+	rootCmd.PersistentFlags().StringVarP(&mustGatherDirPath, "must-gather-dir-path", "M", "/must-gather", "Must gather directory path")
```

Comment:
ditto

Reply:
This is a directory we want to mount and the tool needs to know about. So we get the dir path  at commadline.
How can we do it better?

Reply:
This is fine. The default is less fine: `/must-gather` is outside of the file system hierarchy standard. Can't we just use a relative path? `must-gather` (not lack of heading '/')

Reply:
We can, this will be used to mount the actual must-gather dir from the host. The default value can be anything. A dir inside /usr/local/bin is a good place like any other. We are going to override it anyway.

Reply:
yeah, if we stricly focus on the containerized use case it makes sense. Not a big fan fan but not a big deal either.

Reply:
I can change it to `must-gather` if that looks better.